packages/wrangler/templates/startDevWorker/InspectorProxyWorker.ts

```diff
@@ -212,47 +259,48 @@ export class InspectorProxyWorker implements DurableObject {
 		}
 	};
 
-	handleProxyControllerIncomingMessage = (event: MessageEvent) => {
-		assert(
-			typeof event.data === "string",
-			"Expected event.data from proxy controller to be string"
-		);
-
-		const message: InspectorProxyWorkerIncomingWebSocketMessage = JSON.parse(
-			event.data
-		);
-
-		this.sendDebugLog("handleProxyControllerIncomingMessage", event.data);
-
-		switch (message.type) {
-			case "reloadComplete": {
-				this.proxyData = message.proxyData;
-
-				this.reconnectRuntimeWebSocket();
-
-				break;
-			}
-			default: {
-				assertNever(message.type);
-			}
-		}
-	};
-
+	runtimeAbortController = new AbortController();
 	runtimeKeepAliveInterval: number | null = null;
-	reconnectRuntimeWebSocket() {
+	async reconnectRuntimeWebSocket() {
 		assert(this.proxyData, "Expected this.proxyData to be defined");
 
 		this.sendDebugLog("reconnectRuntimeWebSocket");
 
+		this.runtimeAbortController.abort();
+		this.runtimeAbortController = new AbortController();
 		this.websockets.runtimeDeferred = createDeferred<WebSocket>(
```

Comment:
Assuming it will be closed when the runtime is terminated

Reply:
In fact, it was being closed further down after the next connection had been established. I'm mentioning it here because, to address your other comment, I am changing it to close here as you suggested in this comment